cmd-kill-server.c

```diff
@@ -54,8 +55,11 @@ const struct cmd_entry cmd_start_server_entry = {
 static enum cmd_retval
 cmd_kill_server_exec(struct cmd *self, __unused struct cmdq_item *item)
 {
-	if (cmd_get_entry(self) == &cmd_kill_server_entry)
+	if (cmd_get_entry(self) == &cmd_kill_server_entry) {
+		int status;
 		kill(getpid(), SIGTERM);
+		wait(&status);
```

Comment:
This doesn't do what you think it does. You can't wait for the current process to terminate. The `wait` function waits for a child process to exit. And it will return when any, random, child process terminates. Not when all of them have terminated.

Reply:
I see, maybe using `job_check_died(pid_t pid, int status)`? It'd be useful to make sure all the children are dead before returning `kill-server`.
But it uses a pid that is supposed to be dead and kernel can reassign it, leading to a false negative.

Reply:
The `kill(getpid(), SIGTERM)` is telling the current tmux daemon to terminate. See the logic in `server_signal()` and `server_send_exit()`. You're approaching this the wrong way. The check for whether the server is dead belongs in the tmux client code, not the server code. The tmux client that forwards the `kill-server` to the daemon should, arguably, check that the daemon is actually dead and only finish (i.e., exit) when that is true.

Reply:
Indeed, I'll re-open a merge request once it's done and tested. Thank you!